core/src/main/java/hudson/model/ParametersAction.java

```diff
@@ -231,9 +238,23 @@ public ParametersAction createUpdated(Collection<? extends ParameterValue> overr
     @Nonnull
     public ParametersAction merge(@CheckForNull ParametersAction overrides) {
         if (overrides == null) {
-            return new ParametersAction(parameters);
+            ParametersAction parametersAction = new ParametersAction(parameters);
+            loadSafeParameters();
```

Comment:
Possible, If I do with the constructor approach then there would be no need through.